src/botnet.rs

```diff
@@ -0,0 +1,105 @@
+use parse_args::AppConfig;
+use rand::{thread_rng, Rng};
+use utils::{csleep, dprint};
+use yansi::Paint;
+
+pub fn run(appconfig: &AppConfig) {
+    let mut rng = thread_rng();
+    let clusters = {
+        let mut ret = vec![];
+        let num_clusters = rng.gen_range(8, 16 + 1);
+        for _ in 0..num_clusters {
+            let num_nodes = rng.gen_range(100, 200 + 1);
+            ret.push(num_nodes);
+        }
+        ret
+    };
+    let mut onlines = vec![false; clusters.len()];
+    let size: usize = clusters.iter().sum();
+
+    let mut connected = 0;
+
+    while connected <= size {
+        dprint(
+            format!(
+                "\rEstablishing connections: {connected:4}/{size:4}",
+                connected = connected,
+                size = size
+            ),
+            0,
+        );
+        connected += 1;
+        csleep((rng.gen_range(0f64, 1.).powi(50) * 50.) as u64);
+    }
+    dprint("\n", 0);
+
+    csleep(300);
+
+    for (i, nodes) in clusters.iter().enumerate() {
+        dprint(
+            format!(
+                "  Cluster #{i:02} ({nodes:3} nodes)\n",
+                i = i,
+                nodes = nodes
+            ),
+            10,
+        );
+        csleep(100);
+        if appconfig.should_exit() {
+            return;
+        }
+    }
+
+    loop {
+        dprint(format!("\u{001b}[{line_up:}A", line_up = onlines.len()), 0);
```

Comment:
Wouldn't it be nice if you made some cursor movement utilities instead and used them in your code? Those would be nice additions to `utils.rs` and would make this a lot more readable.